docs/auth.md

```diff
@@ -4,35 +4,225 @@ linkTitle: "Authentication"
+  - [`basic-auth` for Git](#basic-auth-for-git)
+  - [`ssh-auth` for Git](#ssh-auth-for-git)
+  - [`basic-auth` for Docker](#basic-auth-for-docker)
+
+## Overview
+
+Tekton supports authentication via the Kubernetes first-class `Secret` types listed below.
+A `Run` gains access to these `Secrets` through its associated `ServiceAccount`.
+
+<table>
+	<thead>
+		<th>Git</th>
+		<th>Docker</th>
+	</thead>
+	<tbody>
+		<tr>
+			<td><code>kubernetes.io/basic-auth</code><br>
+				<code>kubernetes.io/ssh-auth</code>
+			</td>
+			<td><code>kubernetes.io/basic-auth</code><br>
+				<code>kubernetes.io/dockercfg</code><br>
+				<code>kubernetes.io/dockerconfigjson</code>
+			</td>
+	</tbody>
+</table>	
+
+Since neither Git nor Docker can consume these `Secrets` in their native format, Tekton must first
+convert them as follows:
+
+- For Git, convert into a `.gitconfig` file.
+- For Docker, convert into a `~/.docker/config.json` file.
+
+Additionally, even though each of these entities supports multiple credentials for multiple
+domains, Tekton must first blend those credentials into a single canonical keyring.
+
+To consume these `Secrets`, Tekton performs credential initialization within every `Pod` it instantiates,
+before retrieving any associated `PipelineResources`. During credential initialization, Tekton accesses
+each `Secret` associated with the `Run` and aggregates all of those `Secrets` into their respective
+files in `$HOME/tekton/home`.
+
+## Understanding credential selection
+
+A `Run` might require multiple types of authentication. For example, a `Run` might require access to
+multiple private Git and Docker repositories. You can use credential annotation keys to direct Tekton to
+use the appropriate `Secret` for each associated resource. A credential annotation key must begin with
+`tekton.dev/git-` or `tekton.dev/docker-` and its value is the URL of the host for which you want Tekton
+to use that credential.
+
+In this example, Tekton uses a `basic-auth` (username/password pair) `Secret` to access
+Git repositories at `github.com` and `gitlab.com` as well as Docker repositories at `gcr.io`:
 
-This document defines how authentication is provided during execution of a
-`TaskRun` or a `PipelineRun` (referred to as `Runs` in this document).
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/git-0: https://github.com
+    tekton.dev/git-1: https://gitlab.com
+    tekton.dev/docker-0: https://gcr.io
+type: kubernetes.io/basic-auth
+stringData:
+  username: <cleartext non-encoded>
+  password: <cleartext non-encoded>
+```
+
+And in this example, Tekton uses an `ssh-auth` `Secret` to access Git repositories
+at `github.com` only:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/git-0: github.com
+type: kubernetes.io/ssh-auth
+data:
+  ssh-privatekey: <base64-encoded>
+  # This is non-standard, but its use is encouraged to make this more secure.
```

Comment:
The known_hosts section is probably better to remove from this example. It isn't related to the rest of the way the Secret is setup, and might be distracting.

Reply:
Deleted the `known_hosts` field.